core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/providers/TitleBlockAdapterFactory.java

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ * Copyright (c) 2020 THALES GLOBAL SERVICES.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *  
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.diagram.helpers.providers;
+
+import org.eclipse.emf.common.notify.Adapter;
+import org.eclipse.emf.common.notify.Notifier;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.sirius.viewpoint.description.DAnnotation;
+import org.eclipse.sirius.viewpoint.description.provider.DescriptionItemProviderAdapterFactory;
+import org.polarsys.capella.core.diagram.helpers.TitleBlockHelper;
+
+public class TitleBlockAdapterFactory extends DescriptionItemProviderAdapterFactory {
```

Comment:
There's no need to extend DescriptionItemProviderAdapterFactory. You could just pass it directly to the decorator.

Reply:
OK.

Reply:
Is because I kept the original provider for DAnnotation (added this provider only if the DAnnotation is for aTitle Block).